presto-main/src/main/java/com/facebook/presto/execution/resourceGroups/InternalResourceGroupManager.java

```diff
@@ -116,16 +120,19 @@ public InternalResourceGroupManager(
             ServerConfig serverConfig)
     {
         requireNonNull(queryManagerConfig, "queryManagerConfig is null");
+        this.queryManagerConfig = requireNonNull(queryManagerConfig, "queryManagerConfig is null");
         this.exporter = requireNonNull(exporter, "exporter is null");
         this.configurationManagerContext = new ResourceGroupConfigurationManagerContextInstance(memoryPoolManager, nodeInfo.getEnvironment());
-        this.legacyManager = requireNonNull(legacyManager, "legacyManager is null");
-        this.configurationManager = new AtomicReference<>(cast(legacyManager));
+        ResourceGroupConfigurationManager initializingConfigurationManager = new InitializingConfigurationManager();
```

Comment:
Do we need this local variable? initializingConfigurationManager
We should be able to initialize member variable initializingConfigurationManager directly?